boa/src/builtins/regexp/mod.rs

```diff
@@ -714,4 +715,37 @@ impl RegExp {
 
         Ok(result)
     }
+
+    /// `RegExp.prototype[ @@search ]( string )`
+    ///
+    /// This method executes a search for a match between a this regular expression and a string.
+    ///
+    /// More information:
+    ///  - [ECMAScript reference][spec]
+    ///  - [MDN documentation][mdn]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-regexp.prototype-@@search
+    /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/@@search
+    pub(crate) fn search(this: &Value, args: &[Value], context: &mut Context) -> Result<Value> {
```

Comment:
This should be fixed. The spec only notes that `this` should be an `object`, but I also checked if it is an `regexp` as that seems to make sense.

Reply:
Sorry, but we actually shouldn't do that, this function can be called with other objects and shouldn't fail because of it.
```js
let re = /a/;
let s = "bbabb";
let o = new Object();
o.exec = (a) => {
    console.log(a);
    return null;
}
re[Symbol.search].call(o, s); // returns -1
// logs "bbabb"
```
I know it can seem weird, but pretty much anything can be changed at runtime in JS.

Reply:
I have to take the spec a little bit more literal it seems.

Thanks for the explanation. Should be good now, two more 262 tests are passing for me.

Reply:
I updated the conformance numbers :)